File: examples/cli_modular/commands/greet.js
// commands/greet.js — Print a greeting.

import { greeting } from "./../lib/fmt.js";

export function run(ctx) {
    const name = ctx.args[0];
    if (!name) {
        ctx.stderr.write("greet: missing NAME argument\n");
        return 2;
    }
    ctx.stdout.write(greeting(name) + "\n");
    return 0;
}

File: examples/cli_modular/commands/greet.lua
-- commands/greet.lua - Print a greeting.
--
-- Convention: each command file exports a single `run(ctx)` function.
-- ctx is the same shape app.main receives: args / env / stdin / stdout
-- / stderr. The return value is the process exit code (nil → 0).

local fmt = require("./../lib/fmt")

local M = {}

function M.run(ctx)
    local name = ctx.args[1]
    if not name then
        ctx.stderr:write("greet: missing NAME argument\n")
        return 2
    end
    ctx.stdout:write(fmt.greeting(name) .. "\n")
    return 0
end

return M
